src/System.Runtime.Extensions/src/System/Environment.Unix.cs

```diff
@@ -358,11 +358,11 @@ private static int FindAndParseNextNumber(string text, ref int pos)
             return num;
         }
 
-        public static int ProcessorCount => (int)Interop.Sys.SysConf(Interop.Sys.SysConfName._SC_NPROCESSORS_ONLN);
+        public static int ProcessorCount => CheckedSysConf(Interop.Sys.SysConfName._SC_NPROCESSORS_ONLN);
 
         public static string SystemDirectory => GetFolderPathCore(SpecialFolder.System, SpecialFolderOption.None);
 
-        public static int SystemPageSize => (int)Interop.Sys.SysConf(Interop.Sys.SysConfName._SC_PAGESIZE);
+        public static int SystemPageSize => CheckedSysConf(Interop.Sys.SysConfName._SC_PAGESIZE);
```

Comment:
Why here, and not in the other 2 places we call Sys.SysConf?

However I'm unconvinced that it makes sense to add checking to calls to SysConf, just because a particular invalid combination of our binaries caused an invalid return value from it as opposed to many other kinds of possible misbehavior. Do we have any envidence that this function is particularly interesting or error prone in normal situations?

Reply:
> Why here, and not in the other 2 places we call Sys.SysConf?

Because Environment.ProcessorCount is super common, used in a lot of initialization code (e.g. to size arrays). The others are not. I only changed the other one in this project since why not.

> Do we have any envidence that this function is particularly interesting or error prone in normal situations?

As I noted, I don't think it should ever throw in normal situations. The only documented reason sysconf fails is for an invalid input, and we should be passing in valid input.

Reply:
(If you're really against this, I can also just close it. But the effect of the method returning -1 often isn't noticed until later with an unrelated exception at an unrelated location and it can be difficult to determine the root cause.)

Reply:
Do we have any idea why this particular function is the victim when binaries are mismatched? Generally when there are mismatches all kinds of weird failures can happen. @janvorli

Reply:
See the linked issue: https://github.com/dotnet/corefx/issues/24663#issuecomment-338102110

Reply:
I see.